utils/threshAlgos.py

```diff
@@ -626,10 +667,45 @@ def calibrateThrDAC(inputFile,fitRange="0,255",listOfVFATs=None,noLeg=False,outp
                 dict_ScurveSigmaVsThrDac_BoxPlot[vfat].SetXTitle(thrDacName)
                 dict_ScurveSigmaVsThrDac_BoxPlot[vfat].SetYTitle("Scurve Sigma #left(fC#right)")
 
+            if vfat > -1:    
+                scurveFitDataThisVfat = scurveFitData[scurveFitData["vfatN"] == vfat]
+            else:
+                scurveFitDataThisVfat = scurveFitData
+
+            #define histograms which will hold the per-VFAT scurve mean and scurve sigma distributions
+            #when there is no data, we create a dummy histogram which will not be filled with anything
+            if len(scurveFitDataThisVfat) != 0:
+                thisVFAT_ThreshMean = np.mean(scurveFitDataThisVfat["threshold"])
+                thisVFAT_ThreshStd = np.std(scurveFitDataThisVfat["threshold"])
+                thisVFAT_ENCMean = np.mean(scurveFitDataThisVfat["noise"])
+                thisVFAT_ENCStd = np.std(scurveFitDataThisVfat["noise"])
+            else:    
+                thisVFAT_ThreshMean = 0 #dummy value
+                thisVFAT_ThreshStd = 1 #dummy value
+                thisVFAT_ENCMean = 0 #dummy value
+                thisVFAT_ENCStd = 1 #dummy value
+                
+            histThresh = r.TH1F("scurveMean_vfat%i"%vfat,"VFAT %i;S-Curve Mean #left(fC#right);N"%vfat, 
+                                40, thisVFAT_ThreshMean - 5. * thisVFAT_ThreshStd, thisVFAT_ThreshMean + 5. * thisVFAT_ThreshStd )
+            histENC = r.TH1F("scurveSigma_vfat%i"%vfat,"VFAT %i;S-Curve Sigma #left(fC#right);N"%vfat,
+                                 40, thisVFAT_ENCMean - 5. * thisVFAT_ENCStd, thisVFAT_ENCMean + 5. * thisVFAT_ENCStd )
+                
+            #fill histograms with the scurve means and the scurve sigmas that pass the quality cuts
+            #leave TH1Fs and TGraphs empty for ARM DAC points with too few unmasked channels
+            from gempython.gemplotting.utils.anaInfo import numOfGoodChannelsMin
+            if len(scurveFitDataThisVfat) >= numOfGoodChannelsMin:
```

Comment:
numOfGoodChannelsMin defaults to 10; but what if a user had intentionally only scanned 1 channel at each `CFG_THR_ARM_DAC` value?  They would be unable to use your tool in the analysis.  This should be an optional input argument that the user may provide that defaults to 10, and if the input namespace does not have this attribute it should be set to `numOfGoodChannelsMin`.

Reply:
Done. However note that the case of scanning a single channel is not really supported, we are performing Gaussian fits and then applying cuts to the errors of those Gaussian fits. If we would like to support the case of scanning a single channel, I can make another github issue and pull request.